### PR TITLE
Development Sprint 4

### PR DESCRIPTION
**Pull request by Leidse**

### CODE REVIEW

File: src/lidar_mini.cpp [hunk 1]
```diff
@@ -7,30 +7,69 @@
...
 }
 
 int LIDARmini::getDistance() {
-    std::array<char, 7> data = this->getSensorData();
-    return static_cast<int>(data[0]);
-}
\ No newline at end of file
+    std::array<char, 9> distanceRegisters =
+        this->getWantedRegisters(LidarMiniRegisters::Distance); // 0110 0000 because we want the 2 dist registers
+    int dist_L = distanceRegisters[0];                          // dist_L is register 2
```
Comment (ghost) [inline src/lidar_mini.cpp]: dist_L and dist_H are not camalCase.

File: src/lidar_mini.cpp [hunk 2]
```diff
@@ -7,30 +7,69 @@
...
+    int dist_H = distanceRegisters[1];                          // dist_H is register 3
+    return (dist_H << 8) | (dist_L);
+}
+
+int LIDARmini::getStrength() {
+    std::array<char, 9> strengthRegisters =
+        this->getWantedRegisters(LidarMiniRegisters::Strength); // 0001 1000 because we want the 2 strength registers
+    int strength_L = strengthRegisters[0];                      // strength_L is register 4
```
Comment (ghost) [inline src/lidar_mini.cpp]: strength_L and strength_H are not camelCase.

File: src/lidar_mini.hpp [hunk 3]
```diff
@@ -28,18 +30,56 @@ class LIDARmini {
...
+     * 5 : strength_H (the high range strength data)
+     * 6 : reserved register
+     * 7 : signal quality degree
+     * 8 : perity register (sum of all registers minus the perity register itself ofcourse)
+     *
+     * So if registerSetByte = 10100100 the function will return the 2 startbytes, the dist_H and the reserved register.
+     * The function puts these registers in a 9 byte big array this is because there will never be more the 9 registers to get the
+     * data from
+     *
+     * @param
```
Comment (ghost) [inline src/lidar_mini.hpp:64]: The @param doxygen command required a the name of the parameter it is about.
So in this case it should be @param registerSetByte <description> (The description doesn't have to be on the same line.)

File: src/lidar_mini.hpp [hunk 4]
```diff
@@ -28,18 +30,56 @@ class LIDARmini {
...
+    /**
+     * @brief
+     * Makes it so getwantedRegister accepts enums
+     *
+     * @description
+     * This function gets in the enum which is a byte and the casts the byte to a char
+     * and puts that into the getwantedregister function that uses chars.
+     * After that the function will return the array it gets form the getWantedRegisters(char) function
+     *
+     * @param
```
Comment (ghost) [inline src/lidar_mini.hpp:81]: Same as with the previous @param registerSetByte.

File: test/test_main.cpp [hunk 5]
```diff
@@ -1,8 +1,6 @@
 #define CATCH_CONFIG_MAIN // This tells Catch to provide a main() - only do this in one cpp file
 #include "catch.hpp"
 #include "filter.cpp"
-#include "lidar_mini.cpp"
-#include "wrap-hwlib.hpp"
 
 TEST_CASE("Example Test Case") {
```
Comment (ghost) [inline test/test_main.cpp]: Remove default testcase.

**Review by ghost — CHANGES_REQUESTED** [review]
Great documentation, just a few small things that need some changes.

File: src/lidar_mini.cpp (hunk 1 above)
Comment (ghost) [reply src/lidar_mini.cpp]: This has not been addressed.

File: src/lidar_mini.cpp (hunk 2 above)
Comment (ghost) [reply src/lidar_mini.cpp]: This has not been addressed.

File: src/lidar_mini.hpp (hunk 3 above)
Comment (ghost) [reply src/lidar_mini.hpp:64]: This has not been addressed.

File: src/lidar_mini.hpp (hunk 4 above)
Comment (ghost) [reply src/lidar_mini.hpp:81]: This has not been addressed.

**Comment by ghost** [comment]
Some comments have not been addressed yet.

**Comment by SamZand** [comment]
Close in favour for new pr

**Pull request closed without merging**